contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -377,27 +378,21 @@ contract MixinExchangeCore is
     {
         // Revert if fill amount is invalid
         // TODO: reconsider necessity for v2.1
-        require(
-            takerAssetFillAmount != 0,
-            "INVALID_TAKER_AMOUNT"
-        );
-        
+        if (takerAssetFillAmount == 0)
+            rrevert(FillError(orderInfo.orderHash, FillErrorCodes.INVALID_TAKER_AMOUNT));
```

Comment:
Because it would look like `rrequire(takerAssetFillAmount != 0, FillError(...))`.  It's a nicer experience, but it means the revert reason has to be ABI encoded before the condition is checked, prematurely wasting a lot of gas.

Reply:
Maybe. Will check on remix.

Reply:
Yup, `condition || rrevert(Error(stuff));` works. So that's an option if people want a more compact alternative.